### ISSUE TITLE
Running instrumented code: taint detection

### ISSUE THREAD

**Issue by heroinsoul** [Programming-Systems-Lab/phosphor#21]
Hi,

I'm testing Phosphor now using a simple java code that reads from the text file (read.txt) and writes its content to the other file (write.txt). 

```
package phosphorTest;

import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.io.InputStream;
import java.io.OutputStream;
import java.util.ArrayList;

public class Phosphor {
	
	public static ArrayList<Character> c = new ArrayList<Character>();
	
	// Reader code
	//
	public static void readData() throws Exception {
		InputStream is = null;
	      int i;
//	      char c;
	      
	      try{
	         // new input stream created
	         is = new FileInputStream("read.txt");
	         
	         System.out.println("Characters printed:");
	         
	         // reads till the end of the stream
	         while((i=is.read())!=-1)
	         {
	            // converts integer to character
//	            c=(char)i;
	            c.add((char) i);
	            
	            // prints character
	            System.out.print(c);
	            // Thread.sleep(1000);
	         }
	      }catch(Exception e){
	         
	         // if any I/O error occurs
	         e.printStackTrace();
	      }finally{
	         
	         // releases system resources associated with this stream
	         if(is!=null)
	            is.close();
	      }
	}
	
	// Writer code
	//
	public static void writeData() throws Exception {
		
		OutputStream os = null;
		
		try {
	        // create a new OutputStreamWriter
	        os = new FileOutputStream("write.txt");
	        //OutputStreamWriter writer = new OutputStreamWriter(os);
	        
	        for (int i=0; i<c.size(); i++) {
	        		os.write(c.get(i));
	                // flush the stream
	                os.flush();
	        }
		}catch(Exception e){
	         
	         // if any I/O error occurs
	         e.printStackTrace();
		}finally{
	         // releases system resources associated with this stream
	         if(os!=null)
	            os.close();
		}
        
	}
	
	public static void main (String[] args){
		try {
			readData();
			writeData();
		} catch (Exception e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		}
	}
}
```

I assume that the content of the read.txt file is sensitive and I want to be notified if the app tries to copy it. I assume that this code is written by 3rd party developer and I have no access to its content thus can't use Phosphor API to set and get the taints. I therefore use Phosphor's automatic instrumentation and taint tracking.

Following the instructions in README I start from instrumenting my Java JRE (currently jdk-oracle-8). It works fine. I then instrument the code I wrote specifying the -taintSources and -taintSinks options and it also works fine properly loading the sinks and sources.

Content of taint-sources file:
`java/io/InputStream.read()`

Content of taint-sinks file:
`java/io/OutputStream.write()`

Now when I try to run the instrumented code it's executed successfully and I don't see any notification or warning. Seems like I'm missing something. As I understand Phosphor should have detected the write method and either block it or at least throw the exception. I got none. 

Can you please provide more information on how to control if untrusted code does use any of the sink methods with the tainted sources?

Thanks

**Comment by jon-bell** [comment]
Hi Igor,
You need to specify the complete method descriptors, as shown in the sample [taint sources](https://github.com/Programming-Systems-Lab/phosphor/blob/master/Phosphor/src/taint-sources) and [taint sinks](https://github.com/Programming-Systems-Lab/phosphor/blob/master/Phosphor/src/taint-sinks) files (although without the # comment at the start).

The format is the same as [ASM's](http://asm.ow2.org/doc/faq.html#Q7)

Jon

**Comment by heroinsoul** [comment]
Hi Jon,

Thanks for your feedback. I've modified my taint-sources and taint-sinks files accordingly. Now they look like this

Content of taint-sources file:
```
java/io/FileInputStream(Ljava/lang/String;)V
java/io/InputStream.read()I
```

Content of taint-sinks file:
```
java/io/FileOutputStream(Ljava/lang/String;)V
java/io/OutputStream.write(I)V
```

However, now I'm getting **java.lang.ArrayIndexOutOfBoundsException: 1** error when trying to instrument the code. I see that this error was previously [reported](https://github.com/Programming-Systems-Lab/phosphor/issues/2#issuecomment-164480650) as a bug and was fixed but I'm getting it with the latest version of Phosphor. Is there any workaround for it?

Thanks
